core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -283,7 +283,8 @@ class KafkaServer(
           config = config,
           channelName = "forwarding",
           threadNamePrefix = threadNamePrefix,
-          retryTimeoutMs = config.requestTimeoutMs.longValue
+          networkClientRetryTimeoutMs = config.controllerSocketTimeoutMs,
```

Comment:
Pass `networkClientRetryTimeoutMs = config.controllerSocketTimeoutMs` for there to be no behavioral change.